File: parity.py
# Author: Alexandre Brilhante

''' Returns true if m is even, odd otherwise, using greedy approach and divide and conquer. '''

def parity(m):
    if m == 0:
        return True
    elif m == 1:
        return False
    else:
        return parity(m-2)

def parityDC(m):
    if m == 0:
        return True
    elif m == 1:
        return False
    else:
        i = 0
        j = len(str(m))
        while i < j:
            i += 1
            j -= 1
        return parityDC(int(format(i, 'b'))) == parityDC(int(format(j, 'b')))

File: Parity.scala
// Author: Alexandre Brilhante

/* Returns true if the argument is even, false if not, without using addition, substraction, multiplication, division or modulo. */

import scala.collection.mutable.ListBuffer

object Parity {
  def main(args: Array[String]) {
    println(parity(args{0}.toInt))
  }

  def parity(i: Int): Boolean = {
    var even = new ListBuffer[String]()
    var odd = new ListBuffer[String]()
    for (x <- 1 to i) {
      if (even.length == odd.length) even += ""
      else odd += ""
    }
    if (even.length == odd.length) true
    else false
  }
}